node-js-basics-101/node_modules/asynquence-contrib/tests.js

```diff
@@ -0,0 +1,2764 @@
+						done("Hello");
+					},400);
+				},
+				function(done){
+					done("world","it's me!");
+				},
+				ASQ("Ah","Ha!"),
+				function(done){
+					setTimeout(function(){
+						done.fail("News");
+					},250);
+				}
+			)
+			.val(function(){
+				clearTimeout(timeout);
+				var args = ARRAY_SLICE.call(arguments);
+				args.unshift(testDone,label);
+				FAIL.apply(FAIL,args);
+			})
+			.or(function(msg1,msg2,msg3,msg4,msg5){
+				clearTimeout(timeout);
+
+				if (
+					arguments.length === 5 &&
+					typeof msg1 === "undefined" &&
+					msg2 === "Hello" &&
+					ASQ.isMessageWrapper(msg3) &&
+					msg3[0] === "world" &&
+					msg3[1] === "it's me!" &&
+					ASQ.isMessageWrapper(msg4) &&
+					msg4[0] === "Ah" &&
+					msg4[1] === "Ha!" &&
+					typeof msg5 === "undefined"
+				) {
+					PASS(testDone,label);
+				}
+				else {
+					var args = ARRAY_SLICE.call(arguments);
+					args.unshift(testDone,label);
+					FAIL.apply(FAIL,args);
+				}
+			});
+
+			timeout = setTimeout(function(){
+				FAIL(testDone,label + " (from timeout)");
+			},2000);
+		});
+		tests.push(function(testDone){
+			var label = "Contrib Test  #6", timeout, counter = 0;
+
+			ASQ(3)
+			.until(
+				function(done,msg){
+					setTimeout(function(){
+						counter++;
+						if (counter < msg) {
+							done.fail();
+						}
+						else {
+							done(msg,msg * 2);
+						}
+					},10);
+				},
+				function(done,msg1,msg2){
+					setTimeout(function(){
+						counter++;
+						if (counter < msg2) {
+							done.fail();
+						}
+						else {
+							done(msg1,msg2,msg2 * 2);
+						}
+					},10);
+				}
+			)
+			.val(function(msg1,msg2,msg3){
+				if (!(
+					arguments.length === 3 &&
+					msg1 === 3 &&
+					msg2 === 6 &&
+					msg3 === 12 &&
+					counter === 6
+				)) {
+					var args = ARRAY_SLICE.call(arguments);
+					args.unshift(testDone,label);
+					FAIL.apply(FAIL,args);
+				}
+			})
+			.until(
+				function(done,msg){
+					setTimeout(function(){
+						counter++;
+						if (counter > 10) {
+							done["break"](
+								ASQ.messages("Stop","the","madness!")
+							);
+						}
+						else {
+							done.fail();
+						}
+					},10);
+				}
+			)
+			.val(function(){
+				clearTimeout(timeout);
+				var args = ARRAY_SLICE.call(arguments);
+				args.unshift(testDone,label);
+				FAIL.apply(FAIL,args);
+			})
+			.or(function(msg){
+				clearTimeout(timeout);
+
+				if (
+					arguments.length === 1 &&
+					ASQ.isMessageWrapper(msg) &&
+					msg[0] === "Stop" &&
+					msg[1] === "the" &&
+					msg[2] === "madness!" &&
+					counter === 11
+				) {
+					PASS(testDone,label);
+				}
+				else {
+					var args = ARRAY_SLICE.call(arguments);
+					args.unshift(testDone,label);
+					FAIL.apply(FAIL,args);
+				}
+			});
+
+			timeout = setTimeout(function(){
+				FAIL(testDone,label + " (from timeout)");
+			},2000);
+		});
+		tests.push(function(testDone){
+			var label = "Contrib Test  #7", timeout;
+
+			ASQ()
+			["try"](
```

Comment:
__[jshint]__
* Misleading line break before '['; readers may interpret this as an expression boundary.

<!-- Comment by [SideCI](https://sideci.com) -->